dotcom-rendering/src/web/components/PinnedPostLiveness.importable.tsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect } from 'react';
+
+const isServer = typeof window === 'undefined';
+
+const pinnedPost: Element | null = !isServer
+	? window.document.querySelector('[data-gu-marker=pinned-post]')
+	: null;
+
+const pinnedPostCheckBox: Element | null = !isServer
+	? window.document.querySelector('input[name=pinned-post-checkbox]')
+	: null;
+
+/**
+ * hide show more button on pinned post
+ */
+function hideShowMore() {
+	const pinnedPostButton = document.querySelector<HTMLElement>(
+		'#pinned-post-button',
+	);
+	if (pinnedPostButton) pinnedPostButton.style.display = 'none';
+}
+
+export const PinnedPostLiveness = ({}) => {
+	const contentFitsContainer =
+		pinnedPost && pinnedPost.scrollHeight <= pinnedPost.clientHeight;
+	if (contentFitsContainer) hideShowMore();
```

Comment:
I think this is fine but it made me wonder if we might ever have an issue where the dom isn't ready but we try to perform operations against it?

In react you get around this using a `useEffect` which only runs after things have rendered. But in our islands world that won't work. In pure js you use the [DOMContentLoaded](https://developer.mozilla.org/en-US/docs/Web/API/Window/DOMContentLoaded_event) event but maybe that's excessive here?

Of course, none of this is relevant to your PR here and is more of a question around how we carry this pattern forward.

Reply:
Rereading this comment this is more a note to self for me and you can ignore this